stratum/hal/lib/barefoot/bfrt_node.cc

```diff
@@ -195,43 +209,96 @@ ::util::Status BfrtNode::WriteForwardingEntries(
   for (const auto& update : req.updates()) {
     ::util::Status status = ::util::OkStatus();
     switch (update.entity().entity_case()) {
-      case ::p4::v1::Entity::kTableEntry:
-        status = bfrt_table_manager_->WriteTableEntry(
-            session, update.type(), update.entity().table_entry());
+      case ::p4::v1::Entity::kTableEntry: {
+        ::p4::v1::TableEntry table_entry = update.entity().table_entry();
+        if (translation_enabled_) {
+          ASSIGN_OR_RETURN(table_entry,
+                           p4runtime_bfrt_translator_->TranslateTableEntry(
+                               table_entry, true));
+        }
+        status = bfrt_table_manager_->WriteTableEntry(session, update.type(),
+                                                      table_entry);
         break;
+      }
```

Comment:
A thought on how to cut down the boilerplate code by having a generic `TranslatedEntity` function that we call before the `switch` with `update.entity()`. Inside that we check for the flag and either return it unmodified or translated.
What do you think?

Reply:
This could also move `translation_enabled_` and the flag into the translator file.

Reply:
Yes, actually I am working on having generic functions for read/write request and stream messages